### PR TITLE
166187849 Setup and deploy base application to Heroku

### PR DESCRIPTION
**Pull request by alexxsanya**
#### What does this PR do?
- Setup and deploy base application to Heroku

#### Description of Task to be completed?

- add Procfile file to define process type
- add runtime.txt to define python version
- install packages to run the app in production
- add configuration settings to support the app to run in production/staging on heroku

#### How should this be manually tested?
- Kindly visit click [here](https://dashboard.heroku.com/pipelines/96f9db0b-7b9e-4bed-a5b8-31a20b907fa1) to log into the heroku dashboard to view the review apps.
- Using the credentials I shared in our channels for username & password.
Screenshot 2019-06-05 at 17 07 15

- Click on the links below to confirm that the staging and production links are working.
- [staging](https://ah-backend-realers-staging.herokuapp.com/) url
- [production](https://ah-backend-realers.herokuapp.com/)  url

#### Any background context you want to provide?
The project currently had no heroku setup for continuous delivery and with no review apps for the raise PR. so with this PR the integration with set this in our project.

#### What are the relevant pivotal tracker stories?

- Story ID - [166187849](https://www.pivotaltracker.com/story/show/166187849)

#### DEV NOTE
Kindly note that I have changed the SECRET KEY  & DEBUG to pick value from the environment variable; therefore after the branch has been merged on develop, add the following below in you env file : -
- ```export IS_DEBUG_ON=True``` - can be ```True or False```
- ```export SECRET_KEY='anySecretKey'```
Also, remember to do `pip install -r requirements.txt` to install the Heroku dependency packages

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -9,15 +9,15 @@ coreapi==2.3.3
 coreschema==0.0.4
 coverage==4.5.3
 cryptography==2.6.1
+dj-database-url==0.5.0
 Django==2.1
-django-cors-headers==3.0.1
 django-cors-middleware==1.3.1
 django-extensions==2.1.0
-django-filter==2.1.0
```
Comment (jennizalwango) [inline requirements.txt:16]: l would like to know why you deleted `django-cors-headers==3.0.1` and `django-filter==2.1.0
 ` from the requirements.txt file

Comment (alexxsanya) [reply requirements.txt:16]: @jennizalwango, thanks for pointing this out. I guess, I accidently removed these two lines while doing the rebase. let me add them,

File: authors/settings.py [hunk 2]
```diff
@@ -19,11 +20,18 @@
 # See https://docs.djangoproject.com/en/1.11/howto/deployment/checklist/
 
 # SECURITY WARNING: keep the secret key used in production secret!
-SECRET_KEY = '7pgozr2jn7zs_o%i8id6=rddie!*0f0qy3$oy$(8231i^4*@u3'
+SECRET_KEY = os.environ.get('SECRET_KEY') 
 
 # SECURITY WARNING: don't run with debug turned on in production!
-DEBUG = True
-
+DEBUG = os.environ.get('IS_DEBUG_ON')
```
Comment (HabibSentongo) [inline authors/settings.py:26]: I like the fact that you remembered to not leave Debug mode on in deployment environment. I have manually tested following your steps and the credentials you shared on slack. I have confirmed it is successfully deployed.

**Review by HabibSentongo — APPROVED** [review]

File: requirements.txt (hunk 1 above)
Comment (jennizalwango) [reply requirements.txt:16]: Good work @alexxsanya  l have actually followed the steps you outlined and everything so far works

File: authors/settings.py [hunk 3]
```diff
@@ -42,9 +50,12 @@
     'authors.apps.authentication',
     'authors.apps.core',
     'authors.apps.profiles',
+
+    'whitenoise.runserver_nostatic',
```
Comment (neelxie) [inline authors/settings.py:54]: @alexxsanya could you please explain to me what this line does?

**Review by Emmanuelaaron — APPROVED** [review]
Thank you @alexxsanya. Great job!!!
It looks okay following the steps you've given.
Once again thank you.

File: authors/settings.py (hunk 3 above)
Comment (alexxsanya) [reply authors/settings.py:54]: it adds the whitenoise to the APPS. You may find out more details about [whitenoise by clicking here](http://whitenoise.evans.io/en/stable/)

File: authors/settings.py (hunk 2 above)
Comment (alexxsanya) [reply authors/settings.py:26]: thanks @HabibSentongo + the timely review

**Review by neelxie — APPROVED** [review]
Works seamlessly, thank you.

**Review by jennizalwango — APPROVED** [review]

**Pull request merged**